### ISSUE TITLE
Solving this Hard Fail issue

### ISSUE THREAD

**Issue by navinp** [postalhq/postal#653]
Hello,

From time to time, I am getting this error, for which I don't know why. 

In these cases I am forced to use Gmail to contact my customers. 

(all url's are fictive)

> HARD

> FAIL
> Permanent SMTP delivery error when sending to example.com (0.0.0.0)

> 550-Verification failed for <ugykeu@psrp.postalserver.com> 550-The mail server could not deliver mail to ugykeu@psrp.postalserver.com. The account or domain may not exist, they may be blacklisted, or missing the proper dns entries. 550 Sender verify failed

**Comment by navinp** [comment]
Does anyone have a suggestion to this hard fail error:

```
Permanent SMTP delivery error when sending to somedomain (ip address)

550-Requested action not taken: mailbox unavailable 550 invalid DNS MX or A/AAAA resource record
```

**Comment by navinp** [comment]
> It sounds like the original message isn't delivering and the bounce back isn't being delivered(?)
> 
> Does psrp.postalserver.com exist?

I have added:

Type: CNAME
Host: psrp.domain
Value: is an alias of rp.domain 

The wiki does not say anything about adding psrp as CNAME or as any other DNS record...

https://github.com/atech/postal/wiki/Domains-&-DNS-Configuration

**Comment by navinp** [comment]
If suppose I have postal installed on postal.tld

Should there also be a psrp CNAME record ?

Type: cname
host: psrp.postal.tld 
Value: is an alias of rp.postal.tld

**Comment by willpower232** [comment]
The `psrp` should be for the domain you have added to Postal to send from, the `rp` is for Postal itself.

When you add the domain to Postal, it shows you what the SPF and DKIM records should be and at the bottom of that page it also tells you about the Return Path domain which should be along the lines of `psrp.yourdomain.com` pointing to `rp.postal.yourcompany.com`.

**Comment by navinp** [comment]
Okay, if I understand this correctly, _psrp_ shoult NOT be added for the postal.tld server.

PSRP should be added for the customer-domain.

So, if postal.tld needs to send password reset emails and other emails that will be generated by the postal.tld server, would it then still be necessary to add psrp ?

- And would adding it in this case, hurt email delivery? 

In this case psrp is added to postal.tld but also to customer-domain as cname.

**Comment by willpower232** [comment]
The emails that Postal sends are specified in the `smtp:` section of `postal.yml`. If you have used `postal.yourcompany.com` for your Postal server and you want emails it sends like password resets to come from `@postal.yourcompany.com` then it can make sense to create `psrp.postal.yourcompany.com`.

Adding it will never hurt delivery but its always an optional step so if you have trust in the email addresses used by the users of your Postal, you can probably miss it out.

Its very useful to have on the other domains within Postal so you can be aware of and analyse issues with messages being sent.

**Comment by wtom** [comment]
> Does anyone have a suggestion to this hard fail error:
> 
> ```
> Permanent SMTP delivery error when sending to somedomain (ip address)
> 
> 550-Requested action not taken: mailbox unavailable 550 invalid DNS MX or A/AAAA resource record
> ```

Did you find a solution for that? I'm having the same problem sending e-mails to gmx.net and web.de.

**Comment by willpower232** [comment]
If you are seeing this message then there is a minor problem with your DNS records. Have a look at https://www.linode.com/community/questions/10193/550-invalid-dns-mx-or-aaaaa-resource-record for some inspiration on where to look.

**Comment by wtom** [comment]
Yes, i've just fixed it, the rp.postal.example needs to have a mx entry. I've pointed it to postal.example.com. But postal.example.com was a CNAME to mail.example.com. I've changed the mx to mail.example.com and that fixed the problem.

**Comment by navinp** [comment]
Hello,

I checked everything again, and I believe I have solved this for a few recipient domains that previously would give HARDFAIL are now showing SUCCESS. (message delivered) --  thank you.

Even delivering to GMX domain is working now, which has never worked previously.

The only domain still giving problems is ICLOUD dot com

Postal is installed on domain.com (not on a subdomain)

So it's literally like this:

rp.domain.com MX 10 domain.com

I have created a separate MX record:
domain.com MX 10 domain.com

I have checked its rDNS, its IP is pointed to its domain, so rDNS is setup OK.

Any suggestions would be greatly appreciated.

I have contacted Apple, will update if the issue is on their end.
